components/browser/session/src/main/java/mozilla/components/browser/session/engine/middleware/SuspendMiddleware.kt

```diff
@@ -32,18 +32,18 @@ internal class SuspendMiddleware(
         next: (BrowserAction) -> Unit,
         action: BrowserAction
     ) {
-        if (action is EngineAction.SuspendEngineSessionAction) {
-            suspend(context, action)
-        } else {
-            next(action)
+        when (action) {
+            is EngineAction.SuspendEngineSessionAction -> suspend(context, action.sessionId)
+            is EngineAction.KillEngineSessionAction -> suspend(context, action.sessionId)
```

Comment:
nit: can we combine the two: is SuspendAction, KillAction -> suspend ?

Reply:
This doesn't work here unfortunately because then it can't see `action.sessionId`. It will treat `action` as `EngineAction` and then there's no `sessionId` visible.